compiler/tf2nnpackage-value-remote-test/src/TestDataGenerator.cpp

```diff
@@ -0,0 +1,187 @@
+  const circle::Model *circle_model = circle::GetModel(model_data.data());
+  if (circle_model == nullptr)
+  {
+    std::cerr << "ERROR: Failed to load circle '" << circle_file << "'" << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  // load luci module
+  std::unique_ptr<luci::Module> module = luci::Importer().importModule(circle_model);
+  luci_interpreter::Interpreter interpreter(module.get());
+
+  // create random data generator
+  std::random_device rd;  // used to obtain a seed for the random number engine
+  std::mt19937 gen(rd()); // standard mersenne_twister_engine seeded with rd()
+  std::normal_distribution<float> distrib(0.0f, 2.0f); // mean(0.0f), stddev(2.0f)
+
+  /**
+   *  HDF5 layout is like below
+   *
+   *  GROUP "/"
+   *   ㄴGROUP "name"
+   *     ㄴATTRIBUTE "0"
+   *       ㄴDATA (0): "input_01:0"
+   *     ㄴATTRIBUTE "1"
+   *       ㄴDATA (0): "input_02:0"
+   *   ㄴGROUP "value"
+   *     ㄴDATASET "0"
+   *       ㄴDATA ...
+   *     ㄴDATASET "1"
+   *       ㄴDATA ...
+   */
+  // create random data and dump into hdf5 file
+  H5::H5File input_file{prefix + ".input.h5", H5F_ACC_TRUNC};
+  std::unique_ptr<H5::Group> input_name_group =
+      std::make_unique<H5::Group>(input_file.createGroup("name"));
+  std::unique_ptr<H5::Group> input_value_group =
+      std::make_unique<H5::Group>(input_file.createGroup("value"));
+
+  H5::H5File output_file{prefix + ".expected.h5", H5F_ACC_TRUNC};
+  std::unique_ptr<H5::Group> output_name_group =
+      std::make_unique<H5::Group>(output_file.createGroup("name"));
+  std::unique_ptr<H5::Group> output_value_group =
+      std::make_unique<H5::Group>(output_file.createGroup("value"));
+
+  uint32_t input_index = 0;
+  for (uint32_t g = 0; g < circle_model->subgraphs()->size(); g++)
+  {
+    const auto input_nodes = loco::input_nodes(module->graph(g));
+    for (const auto &node : input_nodes)
+    {
+      const auto *input_node = dynamic_cast<const luci::CircleInput *>(node);
+      std::string name = input_node->name();
+      if (name.find(":0") == std::string::npos)
+        name += ":0";
+
+      // create attribute
+      H5::DataSpace name_dataspace(H5S_SCALAR);
+      H5::StrType name_datatype(H5::PredType::C_S1, name.size());
+
+      auto name_attr = input_name_group->createAttribute(std::to_string(input_index), name_datatype,
+                                                         name_dataspace);
+
+      name_attr.write(name_datatype, name);
+
+      // create value
+      std::vector<hsize_t> dims(input_node->rank());
+      for (uint32_t d = 0; d < input_node->rank(); d++)
+      {
+        dims.at(d) = input_node->dim(d).value();
+        assert(dims.at(d) >= 0);
+      }
+      auto dataspace = std::make_unique<H5::DataSpace>(dims.size(), dims.data());
+      auto dataset = std::make_unique<H5::DataSet>(input_file.createDataSet(
+          "value/" + std::to_string(input_index), H5::PredType::NATIVE_FLOAT, *dataspace));
```

Comment:
There can be non-float inputs. It would be better to set the datatype based on the type of `input_node`.

Reply:
@jinevening I'll add TODO.